keps/sig-node/4603-tune-crashloopbackoff/README.md

```diff
@@ -0,0 +1,1431 @@
+* Quantify and expose risks to node stability resulting from decreased and/or
+  hetereogeneous backoff configurations
+* Empirically derive defaults and allowable thresholds that approximate current
+  node stability
+* Provide a simple UX that does not require changes for the majority of
+  workloads
+
+### Non-Goals
+
+<!--
+What is out of scope for this KEP? Listing non-goals helps to focus discussion
+and make progress.
+-->
+
+* This effort is NOT intending to support fully user-specified configuration, to
+  cap risk to node stability
+* This effort is purposefully NOT implementing more complex heuristics by
+  kubelet (e.g. system state, workload specific factors, or by detecting
+anomalous workload behavior) to focus on better benchmarking and observability
+  and address common use cases with easily modelled changes first
+* This effort is NOT changing the independent backoff curve for image pulls
+
+## Proposal
+
+<!--
+This is where we get down to the specifics of what the proposal actually is.
+This should have enough detail that reviewers can understand exactly what
+you're proposing, but should not include things like API designs or
+implementation. What is the desired outcome and how do we measure success?.
+The "Design Details" section below is for the real
+nitty-gritty.
+-->
+
+This design seeks to incorporate a two-pronged approach:
+
+1. Change the existing initial value for the backoff curve to stack more retries
+   earlier for all restarts (`restartPolicy: OnFailure` and `restartPolicy:
+   Always`)
+2. Provide a `restartPolicy: Rapid` option to configure even faster restarts for
+specific Pod/Container (particularly sidecar containers), regardless of exit
+code, reducing the max wait to 1 minute
+
+In addition, part of the alpha period will be dedicated entirely to
+   systematically stress testing kubelet and API Server with different
+   distributions of workloads utilizing the new backoff curves. Therefore, this
+KEP requires instrumentation of enhanced visibility into pod restart behavior to
+enable Kubernetes benchmarking during the alpha phase. During the benchmarking
+period of alpha, kubelet memory and CPU, API server latency, and pod restart
+latency will be observed and analyzed to define the maximum allowable
+restart rate for fully saturated nodes.
+
+Longer term, these
+metrics will also supply cluster operators the data necessary to better analyze
+and anticipate the change in load and node stability as a result of upgrading to
+these changes.
+
+
+#### Existing backoff curve change: front loaded decay
+
+This KEP proposes changing the existing backoff curve to load more restarts
+earlier by changing the initial value of the exponential backoff. A number of
+alternate initial values are modelled below, until the 5 minute cap would be
+reached. This proposal suggests we start with a new initial value of 1s, and
+analyze its impact on infrastructure during alpha.
+
+![](todayvs1sbackoff.png)
```

Comment:
Nit: but applies to most of the graphs like this one, can you add labels for both the axis ?